classes/OccurrenceLabel.php

```diff
@@ -216,287 +210,339 @@ public function getLabelArray($occidArr, $speciesAuthors = false){
+						if (isset($fieldArr['suffix']) && $fieldArr['suffix']) {
+							$fieldDivStr .= '<span class="' . $fieldName . 'Suffix"' . (isset($fieldArr['suffixStyle']) ? ' style="' . $fieldArr['suffixStyle'] . '"' : '') . '>' . $fieldArr['suffix'] . '</span>';
 						}
 						$fieldDivStr .= '</span>';
 						$cnt++;
 					}
 				}
-				if($fieldDivStr) $outStr .= '<div class="field-block'.(isset($bArr['className'])?' '.$bArr['className']:'').'"'.(isset($bArr['style'])?' style="'.$bArr['style'].'"':'').'>'.$fieldDivStr.'</div>';
+				if ($fieldDivStr) $outStr .= '<div class="field-block' . (isset($bArr['className']) ? ' ' . $bArr['className'] : '') . '"' . (isset($bArr['style']) ? ' style="' . $bArr['style'] . '"' : '') . '>' . $fieldDivStr . '</div>';
 			}
 		}
 		return $outStr;
 	}
 
-	private function getDivBlock($divArr,$occArr){
+	private function getDivBlock($divArr, $occArr) {
 		$contentStr = '';
-		if(array_key_exists('blocks', $divArr)) $contentStr = $this->getLabelBlock($divArr['blocks'],$occArr);
-		elseif(array_key_exists('content', $divArr)) $contentStr = $divArr['content'];
-		if($contentStr){
+		if (array_key_exists('blocks', $divArr)) $contentStr = $this->getLabelBlock($divArr['blocks'], $occArr);
+		elseif (array_key_exists('content', $divArr)) $contentStr = $divArr['content'];
+		if ($contentStr) {
 			$attrStr = '';
-			if(isset($divArr['className'])) $attrStr .= 'class="'.$divArr['className'].'"';
-			if(isset($divArr['style']) && $divArr['style']) $attrStr .= 'style="'.$divArr['style'].'"';
-			return '<div '.trim($attrStr).'>'.$contentStr.'</div>'."\n";
+			if (isset($divArr['className'])) $attrStr .= 'class="' . $divArr['className'] . '"';
+			if (isset($divArr['style']) && $divArr['style']) $attrStr .= 'style="' . $divArr['style'] . '"';
+			return '<div ' . trim($attrStr) . '>' . $contentStr . '</div>' . "\n";
 		}
 		return '';
 	}
 
-	public function getLabelFormatByID($labelCat, $labelIndex){
-		if(is_numeric($labelIndex)){
-			if($labelCat == 'global'){
-				if(file_exists($GLOBALS['SERVER_ROOT'].'/content/collections/reports/labeljson.php')){
-					include($GLOBALS['SERVER_ROOT'].'/content/collections/reports/labeljson.php');
-					if(isset($LABEL_FORMAT_JSON)){
-						if($labelFormatArr = json_decode($LABEL_FORMAT_JSON,true)){
-							if(isset($labelFormatArr['labelFormats'][$labelIndex])){
-								return $labelFormatArr['labelFormats'][$labelIndex];
-							}
-							else $this->errorArr[] = 'ERROR returning global format: index does not exist';
-						}
-						else $this->errorArr[] = 'ERROR returning global format: issue parsing JSON string';
-					}
-					else $this->errorArr[] = 'ERROR returning global format: $LABEL_FORMAT_JSON does not exist';
+	private function transferFromPhpToDynamicProperties() {
+		$status = false;
+		if (file_exists($GLOBALS['SERVER_ROOT'] . '/content/collections/reports/labeljson.php')) {
+			include($GLOBALS['SERVER_ROOT'] . '/content/collections/reports/labeljson.php');
```

Comment:
If there's already a security vulnerability in this file, it will be run here. So, this shouldn't be run until the php file is vetted.

Reply:
Or, alternativley, just delete the php file and start from scratch, but the end user would lose their save profiles in this case.

Reply:
Fixed in 8cb1d04f7.